add_route.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+#
+# add_route.sh
+# Script to setup persistant TEDAPI Routing for Powerwall Dashboard
+# Version: 1.0.0
+# By Scott Hibbard  - 2024-09-14
+#
+
+CRONTAB="/var/spool/cron/crontabs/root"
+PW_IP=""
+SCRIPT_NAME="TDAPI_routing"
+DIR="/root/scripts"
+
+echo "Setup script for persistant Powerall Dashboard TEDAPI Interface network routing"
+echo "-------------------------------------------------------------------------------"
+echo
+echo "This script will require root privileges to read & set a startup cron task."
+read -r -p "Do you want to run this script? [Y/n] " response
+if [[ "$response" =~ ^([nN][oO]|[nN])$ ]]; then
+	echo "Cancel"
+	exit 1
+fi
+
+OS=$(uname -s)
+
+if $(ip route | grep -qw "192.168.91.0/24"); then
+	read -r -p "192.168.91.0/24 routing is already in routing table. Still want to run this? [y/N] " response
+	if [[ ! "$response" =~ ^([yY][eE][sS]|[yY])$ ]]; then
+		echo "Cancel"
+		exit 1
+	fi
+fi
+
+# Running Linux?
+if [[ "${OS}" == "Linux" ]]; then
+	if ! sudo test -d ${DIR}; then
+		sudo mkdir ${DIR}
+		echo "Created folder ${DIR}."
+	fi
+	
+	if [ -f ${DIR}/${SCRIPT_NAME}.sh ]; then
+		echo "Boot script already exists."
+	else
+	    while [ "$PW_IP" == "" ]; do
+   		    read -p 'Enter Powerwall IP Address: ' PW_IP
+		done
+	    echo ""
+		cat > ${SCRIPT_NAME}.tmp << EOF
+#!/bin/bash
+declare -i i=0
+while (( i < 15 )); do
+    RETURN="\$(ip route add 192.168.91.0/24 via ${PW_IP} 2>&1)"
+    if [ "\$RETURN" != "RTNETLINK answers: File exists" ]; then
+        declare -i i=i+1
+        sleep 1
+    else
+        RETURN="Success"
+        break
+    fi
+done
+
+# Uncomment the lines below to log results of executing this script
+# NOW="\$(date)"
+# echo "\${NOW}: result=\${RETURN}, delay=\${i}" >> ${DIR}/${SCRIPT_NAME}.log
+
+EOF
+	chmod 775 ${SCRIPT_NAME}.tmp
+	sudo chown 0 ${SCRIPT_NAME}.tmp
+	sudo mv ${SCRIPT_NAME}.tmp ${DIR}/${SCRIPT_NAME}.sh
+	fi
+
+	if ! $(sudo cat ${CRONTAB} | grep -qw ${DIR}/${SCRIPT_NAME}.sh); then
+		sudo crontab -u root -l > ${SCRIPT_NAME}".cron"
+		echo "@reboot ${DIR}/${SCRIPT_NAME}.sh" >> ${SCRIPT_NAME}".cron"
+		sudo crontab -u root ${SCRIPT_NAME}".cron"
+		rm ${SCRIPT_NAME}".cron"
+		echo "Cron entry added."
+	else
+		echo "Cron line already exists."
+	echo "Installation complete."
+	fi
+
+else
+    echo "You are running '$OS', which is not supported in this script."
+    echo "Maybe you could add code to support '$OS'!"
```

Comment:
Let's add a code block and some logic for MacOS and Windows:

```bash
# MacOS 
sudo route add -host 192.168.91.1 192.168.0.100 # Temporary 
sudo networksetup -setadditionalroutes Wi-Fi 192.168.91.1 255.255.255.255 192.168.0.100 # Persistent

# Windows - Using persistence flag - Administrator Shell
sudo route -p add 192.168.91.1 mask 255.255.255.255 192.168.0.100
```

Reply:
Maybe something like...

```bash
#!/bin/bash

OS=$(uname -s)

# Detect OS and run commands accordingly
if [[ "${OS}" == "Linux" ]]; then
    # Check if running under WSL
    if grep -qi "microsoft" /proc/version; then
        echo "WSL detected - unable to add route automatically."
        echo "You will need to open an Administrator Shell in Windows and run this command: "
        echo "   route -p add 192.168.91.1 mask 255.255.255.255 192.168.0.100"
        echo ""
        exit
    else
        echo "Native Linux detected"
        # ... all the code we have above
        #
    fi
elif [[ "${OS}" == "Darwin" ]]; then
    echo "macOS detected - adding permanent route for Wi-Fi"  # TODO: support for other interfaces
    echo "  sudo networksetup -setadditionalroutes Wi-Fi 192.168.91.1 255.255.255.255 192.168.0.100"
    sudo networksetup -setadditionalroutes Wi-Fi 192.168.91.1 255.255.255.255 192.168.0.100 
    echo ""
    exit
elif [[ "${OS}" =~ MINGW* || "${OS}" =~ CYGWIN* ]]; then
        echo "WSL detected - attempting to add route automatically."
        echo "   route -p add 192.168.91.1 mask 255.255.255.255 192.168.0.100"
        route -p add 192.168.91.1 mask 255.255.255.255 192.168.0.100
        echo ""
        exit
else
    echo "You are running '$OS', which is not supported in this script."
    echo "Maybe you could add code to support '$OS'!"
fi
```

Reply:
Thanks, don't have a MAC, and did not try a Windows install, so couldn't test them.  Will add.